content/source/docs/enterprise/system-overview/architecture.html.md

```diff
@@ -5,24 +5,201 @@ page_title: "Architecture Summary - System Overview - Terraform Enterprise"
 
 # Terraform Enterprise Architecture
 
-This page describes aspects of the architecture of Terraform Enterprise.
+Terraform Enterprise is powered by the same set of services behind HashiCorp’s public SaaS application Terraform Cloud. The sections below present the functional layers of the architecture, how they interact and details pertaining to the corresponding Docker containers involved. References to Docker containers in this document are those you will see when running a `docker ps` on a Terraform Enterprise application server while the application is up.
 
--> **Note:** This page was written for a prior implementation of Terraform Enterprise, and may be outdated. We plan to review our architecture documentation in the near future.
+~> **Advanced:** The content in this document is not required to operate Terraform Enterprise but is provided to help advanced users understand the product in more depth.
 
-## Services
+~> **Note:** Several services described below include the legacy string `ptfe` as part of their names. See [here](https://www.terraform.io/docs/enterprise/index.html#note-about-product-names) for more details.
 
-These are the services used to run Terraform Enterprise. Each service contains a description of what actions it performs, a policy for restarts, impact of failing or degraded performance, and the service's dependencies.
+## Application Layer
+
+The application layer is divided conceptually into two areas, each with specific responsibilities:
+
+- Web Services - Answer requests forwarded by NginX and enqueue jobs for later
+- Background Services - Asynchronously process jobs enqueued by web services
+
+Most requests in Terraform Enterprise follow a similar lifecycle as shown in the diagram below.
+
+![tfe-int-app-layer](assets/tfe-int-app-layer.png)
+
+A web service receives a request and enqueues a background job to perform the task via the Coordination Layer (see below).  The web service then responds immediately to the original request instead of waiting for the job to be finished.
```

Comment:
```suggestion
A web service receives a request and enqueues a background job to perform the task via the Coordination Layer (see below). The web service then responds immediately to the original request instead of waiting for the job to be finished.
```